src/main/java/com/uber/jenkins/phabricator/coverage/JacocoCoverageProvider.java

```diff
@@ -115,29 +113,28 @@ private JacocoBuildAction getJacocoBuildAction() {
         return build.getAction(JacocoBuildAction.class);
     }
 
-    private JacocoPublisher getJacocoPublisher() {
-        Project<?, ?> project = (Project<?, ?>) ((AbstractBuild) build).getProject();
-        return (JacocoPublisher) project.getPublisher(JacocoPublisher.DESCRIPTOR);
-    }
-
     @Override
     public Map<String, List<Integer>> readLineCoverage() {
         JacocoBuildAction jacocoAction = getJacocoBuildAction();
-        JacocoPublisher jacocoPublisher = getJacocoPublisher();
-        if (jacocoAction == null || jacocoPublisher == null) {
+
+        if (jacocoAction == null) {
             return null;
         }
 
         HashMap<String, List<Integer>> lineCoverage = new HashMap<String, List<Integer>>();
 
         String[] includes = null;
-        if (jacocoPublisher.getInclusionPattern() != null) {
-            includes = new String[] {jacocoPublisher.getInclusionPattern()};
-        }
-
         String[] excludes = null;
-        if (jacocoPublisher.getExclusionPattern() != null) {
-            excludes = new String[] {jacocoPublisher.getExclusionPattern()};
+
+        try {
+            Field inclusionsField = JacocoBuildAction.class.getDeclaredField("inclusions");
```

Comment:
This is bound to break on plugin upgrades and is generally unsafe. I would recommend you to not rely on a jacoc publish step and use the upcoming functionality to let the phabricator jenkins plugin directly parse the jacoco xml